### ISSUE TITLE
Need a little help transitioning from flex/bison to jison-lex/jison

### ISSUE THREAD

**Issue by Ravenwater** [zaach/jison#331]
In the flex/bison world, you can write simple text processing utilities. For example, a wc program:

%{
/*
 * word count
 */

var nrchars, nrwords, nrlines;

%}

%%

\n      	++nrchars, ++nrlines;
[^ \t\n] 	++nrwords, nrchars += yyleng;
. 		++nrchars;

%%

main() {
        yylex();
       printf("%d\t%d\t%d\n", nrchars, nrwords, nrlines);
}

---EOF

I have yet to discover how to write these types of lex/yacc tools with jison-lex/jison. Can somebody enlighten me, please?

**Comment by techtonik** [comment]
I wouldn't mind against good jison tutorials too. Or maybe even more human readable way to write rules.

**Comment by yosbelms** [comment]
http://jison.org/

**Comment by Ravenwater** [comment]
http://jison.org/ is wonderful, but I couldn't find any documentation how to replace the C-style bindings to main() that make lex and yacc so productive for building text processing tools.

I do see that the third section of the syntax and grammar files get reproduced just prior to the export. I had hoped that others had used jison-lex/jison as a tool building automation, and thus could bootstrap me, but alas, I'll dive into the code and see if I can make it work.

**Comment by EricGT** [comment]
Since I am just starting out with Jison and using [flex & bison ](https://www.worldcat.org/title/flex-bison/oclc/321016664&referer=brief_results) as a reference which has the word count example and ran into the same problem I am posting this to help others. This is probably not the best way to do it, but it does get one past this example and on to making more progress with Jison.


````
// wordcount.jison

// Based on the example in "flex & bison" by John Levine

// This is a wordcount example.

// Lexer Grammar

%lex

/* Lexer Section 1 : Definitions */


%{
    console.log("In Lexer Definitions section");
%}

%%

/* Lexer Section 2 : Rules */

[a-zA-Z]+
    {
        console.log("In Lexer Rule WORD");
        console.log("Matched: '",this.match,"'");
        return 'WORD';
    }
\n
    {
        console.log("In Lexer Rule LF");
        console.log("Matched: '",this.match,"'");
        return 'LF';
    }
\r
    {
        console.log("In Lexer Rule CR");
        console.log("Matched: '",this.match,"'");
        return 'CR';
    }
<<EOF>>
    {
        console.log("In Lexer Rule EOF");
        console.log("Matched: '",this.match,"'");
        return 'EOF';
    }
.
    {
        console.log("In Lexer Rule SEP");
        console.log("Matched: '",this.match,"'");
        return 'SEP';
    }


%%

/* Lexer Section 3 : User Code */

console.log("In Lexer User Code section");

/lex

// Parser Grammar

/* Parser Section 1 : Definitions */

%{
    /* code block */
    console.log("In Parser Definitions section");
    let myChars = 0;
    let myWords = 0;
    let myLines = 0;
%}

%%

/* Parser Section 2 : Rules */

input
    : sentences eof
    ;

sentences :
      sentence cr lf sentences
    | sentence
    ;

sentence :
      word sep sentence
    | word sep
    | word
    ;

word
    : WORD
        %{
            console.log("In Parser Rule WORD");
            myWords++; myChars += yytext.length;
        %}
    ;

cr  : CR
        %{
            console.log("In Parser Rule CR");
            myChars++;
        %}
    ;

lf  : LF
        %{
            console.log("In Parser Rule LF");
            myChars++; myLines++;
        %}
    ;

sep : SEP
        %{
            console.log("In Parser Rule SEP");
            myChars++;
        %}
    ;

eof : EOF
        %{
            console.log("In Parser Rule EOF");
            myChars++; myLines++;
            console.log("myLines: ", myLines, ", myWords: ", myWords, ", myChars: ", myChars);
        %}
    ;

%%

/* Parser Section 3 : Epilogue */

console.log("In Parser Epilogue section");
````

TL;DR

To help me understand the sections of Jison, I liberally added lots of comments and code sections to see how the user code was getting inserted into the Jison boilerplate code. This helped out because I soon realized that leaving out `return` statements with the lexer actions was causing problems. Also I had to use the parser because I am still learning and have not figured out how to get  just `flex` to work in Jison.

My development environment consist of 
Visual Studio Code v1.5.3
Node.js version  v6.6.0
Jison 0.4.17

Hope this helps you and others.
